metals/src/main/scala/scala/meta/internal/metals/ConfiguredLanguageClient.scala

```diff
@@ -25,22 +25,31 @@ final class ConfiguredLanguageClient(
     extends DelegatingLanguageClient(initial) {
 
   private var clientCapabilities = ClientExperimentalCapabilities.Default
+  private var initializationOptions = InitializationOptions.Default
 
   override def configure(capabilities: ClientExperimentalCapabilities): Unit = {
     clientCapabilities = capabilities
   }
 
+  override def configure(options: InitializationOptions): Unit = {
+    initializationOptions = options
+  }
+
   override def shutdown(): Unit = {
     underlying = NoopLanguageClient
   }
 
   override def metalsStatus(params: MetalsStatusParams): Unit = {
-    if (config.statusBar.isOn || clientCapabilities.statusBarIsOn) {
+    val statusBarIsOn = config.statusBar.isOn ||
+      clientCapabilities.statusBarIsOn ||
+      initializationOptions.statusBarIsOn
+
+    if (statusBarIsOn) {
       underlying.metalsStatus(params)
     } else if (params.text.nonEmpty && !pendingShowMessage.get()) {
-      if (config.statusBar.isShowMessage || clientCapabilities.statusBarIsShowMessage) {
+      if (config.statusBar.isShowMessage || clientCapabilities.statusBarIsShowMessage || initializationOptions.statusBarIsShowMessage) {
```

Comment:
Nit: Can you do the same for this one and the one below and separate this out into a variable like you did for `statusBarIsOn`. Then we can just do checks on `statusBarIsShow` and `statusBarIsLog`.